inspirehep/records/marshmallow/literature/base.py

```diff
@@ -124,3 +129,163 @@ def get_ui_display(self, data):
             return json.loads(ui_display)
         except json.JSONDecodeError:
             return {}
+
+
+class LiteratureESEnhancementV1(LiteratureMetadataSchemaV1):
+    """Elasticsearch serialzier"""
+
+    _created = fields.DateTime(dump_only=True, attribute="created")
+    _updated = fields.DateTime(dump_only=True, attribute="updated")
+    abstract_source_suggest = fields.Method("populate_abstract_source_suggest")
+    author_count = fields.Method("populate_author_count")
+    authors = fields.Method("preprocess_authors")
+    bookautocomplete = fields.Method("populate_bookautocomplete")
+    earliest_date = fields.Method("populate_earliest_date")
+    facet_inspire_doc_type = fields.Method("populate_inspire_document_type")
+    facet_author_name = fields.Method("populate_facet_author_name")
+    id = fields.UUID(dump_only=True)
+    thesis_info = fields.Nested(ThesisInfoSchemaForESV1, dump_only=True)
+
+    def populate_abstract_source_suggest(self, record):
+        """Prepares record for ``abstract_source_suggest`` field."""
+        abstracts = record.get("abstracts", [])
+
+        for abstract in abstracts:
+            source = abstract.get("source")
+            if source:
+                abstract.update({"abstract_source_suggest": {"input": source}})
+        return abstracts
+
+    def populate_earliest_date(self, record):
+        """Prepares record for ``earliest_date`` field."""
+        earliest_date_found = record.get("earliest_date", "")
+        date_paths = [
+            "preprint_date",
+            "thesis_info.date",
+            "thesis_info.defense_date",
+            "publication_info.year",
+            "legacy_creation_date",
+            "imprints.date",
+        ]
+
+        dates = [
+            str(el)
+            for el in chain.from_iterable(
+                [force_list(record.get_value(path)) for path in date_paths]
+            )
+        ]
+        if dates:
+            result = earliest_date(dates)
+            if result:
+                earliest_date_found = result
```

Comment:
`return result`

Reply:
Not really, earliest_date_found keeps default data pulled from record['earliest_date'] and should be used when result is empty. I could jus `return result or earliest_date_found` but I'm trying not to put logic inside return statement.